src/components/todolist/todolist.tsx

```diff
@@ -0,0 +1,292 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Button, Text, Modal, Toast, Icon } from "@innovaccer/design-system";
+import "@innovaccer/design-system/css";
+import { MessageAppearance } from "@innovaccer/design-system/dist/core/common.type";
+import { Dispatch } from "react";
+import React, { useEffect, useState } from "react";
+import { addItem, deleteItem, editItem } from "../../reducer/todolist/action";
+import {
+  IDispatch,
+  IMapDispatchToProps,
+  ItemType,
+  IMapStateToProps,
+  PropType,
+  IState,
+} from "../../interfaces/entities/todolist";
+import { connect } from "react-redux";
+import { Formik, Form, Field, ErrorMessage, useFormik } from "formik";
+import "./styles.css";
+
+const TodoList = (props: PropType) => {
+  const [todoItems, setTodoItems] = useState<Array<ItemType>>([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [toastType, setToastType] = useState<MessageAppearance>();
+  const [actionPerformed, setActionPerformed] = useState(false);
+  const [selectedItem, setSelectedItem] = useState({
+    updatedValue: "",
+    oldValue: "",
+    id: "",
+  });
+
+  useEffect(() => {
+    if (props.todoItemsState) setTodoItems(props.todoItemsState);
+  }, [props.todoItemsState]);
+
+  const CaptureTodo = () => {
+    const [todoValue, setTodoValue] = useState("");
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      setTodoValue(e.target.value);
+    };
```

Comment:
A relatively simpler usage and less bulky code
```suggestion
    const onTextChange = (newText: string) => setTodoValue(newText);
```

Also, update the calling place to this
```
onChange={event => onTextChange(event.target.value)}
```